### PR TITLE
Delete Datastore Table Button

### PR DESCRIPTION
**Pull request by JVickery-TBS**
feat(views): added view and button to delete ds table;

- Added view to delete datastore table.
- Added button to delete a resource's datastore table.

Adds a button to the Xloader data page to delete a Resource's datastore table.

### CODE REVIEW

File: ckanext/xloader/templates/xloader/resource_data.html [hunk 1]
```diff
@@ -5,10 +5,24 @@
...
+  {% block delete_ds_button %}
+    <form method="post" action="{{ delete_action }}" class="mb-3 d-inline-block">
+      {{ h.csrf_input() if 'csrf_input' in h }}
+      <a href="{{ h.url_for('xloader.delete_datastore_table', id=pkg.id, resource_id=res.id) }}"
+        class="btn btn-danger pull-left"
+        type="submit"
+        data-module="confirm-action"
+        data-module-with-data=true
+        data-module-content="{{ _('Are you sure you want to delete this DataStore table and Data Dictionary?') }}"
+        >{% block delete_datastore_button_text %}<i class="fa fa-remove"></i>{{ _('Delete DataStore table') }}{% endblock %}</a>
```
Comment (ThrawnCA) [inline ckanext/xloader/templates/xloader/resource_data.html:20]: Probably shouldn't include the word "table" here since it's not usually in the Datastore UI.

Comment (JVickery-TBS) [reply ckanext/xloader/templates/xloader/resource_data.html]: Word "table" has been removed from user visible things here

**Comment by JVickery-TBS** [comment]
@ThrawnCA for this PR, I also reverted the `ckanapi` dependency addition and used `get_action` from the toolkit instead

**Review by ThrawnCA — APPROVED** [review]

**Comment by ThrawnCA** [comment]
Looks good from here.

**Comment by ThrawnCA** [comment]
Ideally there should be a sanity check to ensure that the package ID is correct for the resource ID.

**Review by ThrawnCA — APPROVED** [review]

File: ckanext/xloader/templates/xloader/resource_data.html [hunk 2]
```diff
@@ -5,10 +5,26 @@
...
   {% set action = h.url_for('xloader.resource_data', id=pkg.name, resource_id=res.id) %}
+  {% set delete_action = h.url_for('xloader.delete_datastore_table', id=pkg.id, resource_id=res.id) %}
   {% set show_table = true %}
 
+  {% block delete_ds_button %}
+    {% if res.datastore_active %}
+      <form method="post" action="{{ delete_action }}" class="mb-3 d-inline-block">
+        {{ h.csrf_input() if 'csrf_input' in h }}
+        <a href="{{ h.url_for('xloader.delete_datastore_table', id=pkg.id, resource_id=res.id) }}"
+          class="btn btn-danger pull-left"
```
Comment (ThrawnCA) [inline ckanext/xloader/templates/xloader/resource_data.html:16]: This should use 'pull-right' so the upload button comes first.

File: ckanext/xloader/templates/xloader/resource_data.html [hunk 3]
```diff
@@ -5,10 +5,26 @@
...
+          type="submit"
+          data-module="confirm-action"
+          data-module-with-data=true
+          data-module-content="{{ _('Are you sure you want to delete the DataStore and Data Dictionary?') }}"
+          >{% block delete_datastore_button_text %}<i class="fa fa-remove"></i>{{ _('Delete from DataStore') }}{% endblock %}</a>
```
Comment (ThrawnCA) [inline ckanext/xloader/templates/xloader/resource_data.html:21]: Needs a space between the icon and text.

Also, consider the `fa-trash-alt` icon instead of `fa-remove`.

**Review by ThrawnCA — CHANGES_REQUESTED** [review]
Testing on dev.data.qld.gov.au indicates that the positioning of the Delete button doesn't mesh well with the Upload to Datastore button.

**Review by ThrawnCA — CHANGES_REQUESTED** [review]
After the confirm-action JavaScript runs, the button submits a datastore refresh instead of deletion.

Our fork will fix this with https://github.com/qld-gov-au/ckanext-xloader/pull/74

File: ckanext/xloader/templates/xloader/resource_data.html (hunk 2 above)
Comment (JVickery-TBS) [reply ckanext/xloader/templates/xloader/resource_data.html:16]: should stop using pulls in CKAN really. They truly are the worst thing for CSS layout and ordering like this. I can wrap in flex box and do a flex revers is probably best

Comment (JVickery-TBS) [reply ckanext/xloader/templates/xloader/resource_data.html]: @ThrawnCA okay this should be good not

File: ckanext/xloader/templates/xloader/resource_data.html (hunk 3 above)
Comment (JVickery-TBS) [reply ckanext/xloader/templates/xloader/resource_data.html]: done

**Comment by JVickery-TBS** [comment]
@ThrawnCA button alignment should be good now

File: ckanext/xloader/views.py [hunk 4]
```diff
@@ -14,7 +14,15 @@ def get_blueprints():
 
 @xloader.route("/dataset/<id>/resource_data/<resource_id>", methods=("GET", "POST"))
 def resource_data(id, resource_id):
-    return utils.resource_data(id, resource_id)
+    rows = request.args.get('rows')
```
Comment (ThrawnCA) [inline ckanext/xloader/views.py:17]: This is unrelated.

Comment (JVickery-TBS) [reply ckanext/xloader/views.py:17]: I needed to merge master because of merge conflicts

File: ckanext/xloader/templates/xloader/resource_data.html [hunk 5]
```diff
@@ -5,17 +5,35 @@
 {% block primary_content_inner %}
 
   {% set action = h.url_for('xloader.resource_data', id=pkg.name, resource_id=res.id) %}
+  {% set delete_action = h.url_for('xloader.delete_datastore_table', id=pkg.id, resource_id=res.id) %}
```
Comment (ThrawnCA) [inline ckanext/xloader/templates/xloader/resource_data.html:8]: Testing at dev.data.qld.gov.au indicates that setting a variable at this point in the template is ineffective. It needs to be set inside the block where it's used.

The previous line inadvertently gets away with it because a blank action means "submit to the current URL", which is the desired behaviour.

**Review by ThrawnCA — CHANGES_REQUESTED** [review]

File: ckanext/xloader/templates/xloader/resource_data.html (hunk 5 above)
Comment (JVickery-TBS) [reply ckanext/xloader/templates/xloader/resource_data.html]: @ThrawnCA cool, did not know about variables and blocks scopes.

This should be fixed now.

**Review by ThrawnCA — APPROVED** [review]
